src/metabase/lib/join.cljc

```diff
@@ -831,6 +835,31 @@
                      y (dissoc y ::lib.card/force-broken-id-refs)]
                  (lib.filter/filter-clause (lib.filter.operator/operator-def :=) x y)))]
        (or
+        ;; Metrics: if both the query stage and the joinable are metrics with at least one breakout, and those breakouts
+        ;; have the same type, suggest joining on those breakout columns.
+        (let [source (lib.metric.basics/source-metric query stage-number)
+              target (lib.metric.basics/join->metric query joinable)
+              #_(when (and (= (:lib/type joinable) :metadata/card)
+                                (= (:type joinable) :metric))
+                       (lib.metadata/metric query (:id joinable)))]
+              #?(:cljs (js/console.log "source" source))
+              #?(:cljs (js/console.log "target" target))
+              #?(:cljs (js/console.log "joinable" joinable))
+          ;; Both are metrics, now check that each has a breakout.
+          (let [source-breakout (some->> source (lib.metric.basics/primary-breakout unjoined))
+                target-breakout (some->> target (lib.metric.basics/primary-breakout unjoined))
+                source-type     (:base-type source-breakout)
+                target-type     (:base-type target-breakout)]
+            (when (and source-breakout target-breakout
+                       source-type target-type
+                       (= source-type target-type))
+              ;; Everything lines up, so return the suggested condition.
+              ;; If the source has a temporal bucket applied, overwrite the target's bucket to match.
+              (let [source-bucket   (lib.temporal-bucket/temporal-bucket source-breakout)
```

Comment:
I think we should add `:default` if missing? Joining on an exact timestamp is probably always a mistake unless they are the same field.

Reply:
What about non-temporal breakouts?

Reply:
Right, only temporal breakouts would get default